src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -29,7 +29,7 @@ internal sealed class NormStr
         private NormStr(ReadOnlyMemory<char> str, int id, uint hash)
         {
             Contracts.Assert(id >= 0 || id == -1 && str.IsEmpty);
-            Value = str;
+            Value = new ReadOnlyMemory<char>(str.ToString().ToCharArray());
```

Comment:
This is Proposal 1

Reply:
I am just realizing that one downside of this proposal would be that when finding n-grams bigger than 1, or char-grams, this proposal would take a lot of memory unnecessarily repeating strings.

Reply:
💭 The call to `ToCharArray()` is unnecessary right?

https://github.com/dotnet/runtime/blob/71d7e6cd72e9d4154f595cc43adac4b66ca53e48/src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs#L164-L173

Reply:
Thanks for the comment @sharwell 

In fact I am just realizing that it might be better to simply use
`Value = str.ToString().AsMemory();`

Reply:
So I've just had a conversation with @justinormont and he does prefer this proposal instead of the other one, since the other one would only work for pipelines that use TextNormalizer, but not every pipeline does that.

He does have some other comments on how this proposal should be done... The most relevant which is that this "str.ToString().AsMemory()" (i.e., creating a new ReadOnlyMemory<char> so that the reference to the whole row is lost) should be left as an option for the user. This would imply adding a flag to several methods, up to the FeturizeText method, and adding the corresponding documentation about the option. So what do people think?

Reply:
Can you review my proposal 3 below too?

Reply:
Yeah, so I would replace proposal 1 with proposal 3, since @justinormont also thought what is done in proposal 3 would be better.

Still, the question remains about if we should make it optional what is done in proposal 3, having to add a new flag up to FeaturizeText.

Reply:
You can find Proposal 3 [here](https://github.com/dotnet/machinelearning/pull/4576/files#r357871260). And I've also updated the first post of this issue with the results of that proposal (which were pretty much the same as Proposal 1 for this specific case).

Reply:
So Proposal 3 was chosen. So I will resolve this thread.